# HG changeset patch
# User Sahil Yadav <yadavsahil5198@gmail.com>
# Date 1516700723 -3600
#      Tue Jan 23 10:45:23 2018 +0100
# Node ID b708857d634e9632a646c63f655e156174f828d9
# Parent  ab2321d4ba03671aff259e89751899c5af6b3466
Correctly display boolean properties default values (bug #52768)

* graphics.in.h (bool_property::bool_property): correctly set default string in
  constructor

* contributors.in: add Sahil Yadav

diff --git a/doc/interpreter/contributors.in b/doc/interpreter/contributors.in
--- a/doc/interpreter/contributors.in
+++ b/doc/interpreter/contributors.in
@@ -419,16 +419,17 @@ Rik Wehbring
 Bob Weigel
 Andreas Weingessel
 Martin Weiser
 Michael Weitzel
 David Wells
 Joachim Wiesemann
 Alexander Wilms
 Georg Wiora
+Sahil Yadav
 Fook Fah Yap
 Sean Young
 Michele Zaffalon
 Serhiy Zahoriya
 Johannes Zarl
 Michael Zeising
 Federico Zenith
 Claudius Zingerli
diff --git a/libinterp/corefcn/graphics.in.h b/libinterp/corefcn/graphics.in.h
--- a/libinterp/corefcn/graphics.in.h
+++ b/libinterp/corefcn/graphics.in.h
@@ -1584,17 +1584,18 @@ class bool_property : public radio_prope
 public:
   bool_property (const std::string& nm, const graphics_handle& h,
                  bool val)
     : radio_property (nm, h, radio_values (val ? "{on}|off" : "on|{off}"))
   { }
 
   bool_property (const std::string& nm, const graphics_handle& h,
                  const char *val)
-    : radio_property (nm, h, radio_values ("on|off"), val)
+    : radio_property (nm, h, radio_values (std::string (val) == "on" ? 
+                                           "{on}|off" : "on|{off}"), val)
   { }
 
   bool_property (const bool_property& p)
     : radio_property (p) { }
 
   bool is_on (void) const { return is ("on"); }
 
   bool_property& operator = (const octave_value& val)
